### PR TITLE
Add support for DataLayer access

### PR DESCRIPTION
**Pull request by sumathin-lang**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description
Adds support for a new private API so Apps can leverage the Data Layer available in the Host

### Main changes in the PR:

1. Add new private dataLayer namespace with the getDataLayerPort API

### Validation performed:

1. Updated the test app to expose the new API
2. Tested against an internal hub to confirm E2E functionality

### Unit Tests added:

Yes

### End-to-end tests added:

<Yes/No>

### Change file added:

Yes

### CODE REVIEW

File: change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Added `{namespace}` capability that will {explain capability}. The capability is still awaiting support in one or most host applications. To track availability of this capability across different hosts see https://aka.ms/capmatrix",
```
Comment (sumathin-lang) [inline change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json:3]: fix this

File: apps/teams-test-app/src/components/privateApis/PrivateAPIs.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {
+  dataLayer,
```
Comment (kerrynf) [inline apps/teams-test-app/src/components/privateApis/PrivateAPIs.tsx:2]: Recommend adding a new file into the test app that contains both your core API test and the capability check. 

You can see here I have just updated my test app component to include the standard format for checking the capability: https://github.com/OfficeDev/microsoft-teams-library-js/pull/2260/files#diff-b23c94c513dc820dfc219abbcdf724085cf88df4c9bcc0bc5b07e84009ed9235

File: change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Added DataLayer capability that will allow Apps to leverage data layer in the Host. The capability is still awaiting support in one or most host applications. To track availability of this capability across different hosts see https://aka.ms/capmatrix",
```
Comment (kerrynf) [inline change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json:3]: Should we change this to be clear that this isn't a generally available capability? I am sure this feature will not be able to be tracked in the public https://aka.ms/capmatrix...

On my telemetry PR my change file reads "Added private MessageChannels capability".

File: apps/teams-test-app/src/components/privateApis/PrivateAPIs.tsx (hunk 2 above)
Comment (sumathin-lang) [reply apps/teams-test-app/src/components/privateApis/PrivateAPIs.tsx]: updated. thanks

File: packages/teams-js/src/private/dataLayer.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    // Send request for dataLayer port, will throw if the request is rejected
+    dataLayerPort = await requestPortFromParentWithVersion(
+      getApiVersionTag(dataLayerVersionNumber, ApiName.PrivateAPIs_DataLayer_GetDataLayerPort),
+      'dataLayer.getDataLayerPort',
```
Comment (sumathin-lang) [inline packages/teams-js/src/private/dataLayer.ts:68]: use enum?

**Review by kerrynf — APPROVED** [review]
Looks good to me. :)

File: packages/teams-js/src/private/dataLayer.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      return dataLayerPort;
+    }
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    // Send request for dataLayer port, will throw if the request is rejected
+    dataLayerPort = await requestPortFromParentWithVersion(
+      getApiVersionTag(dataLayerVersionNumber, ApiName.PrivateAPIs_DataLayer_GetDataLayerPort),
```
Comment (M1Les) [inline packages/teams-js/src/private/dataLayer.ts:67]: Can this have app version as well? Or would we set it in the main renderer?

**Comment by TrevorJoelHarris** [comment]
I believe this is no longer needed, as the PR @kerrynf has made [here](https://github.com/OfficeDev/microsoft-teams-library-js/pull/2277) is the version we'll be going forward with. Feel free to reopen and contact me if that's wrong.

**Pull request closed without merging**